packages/aws-cdk-lib/aws-ec2/lib/cfn-init.ts

```diff
@@ -311,10 +311,17 @@ function deepMerge(target?: Record<string, any>, src?: Record<string, any>) {
       if (target[key] && !Array.isArray(target[key])) {
         throw new Error(`Trying to merge array [${value}] into a non-array '${target[key]}'`);
       }
-      target[key] = Array.from(new Set([
-        ...target[key] ?? [],
-        ...value,
-      ]));
+      if (key != 'command') { // don't deduplicate command arguments
+        target[key] = Array.from(new Set([
+          ...target[key] ?? [],
+          ...value,
+        ]));
+      } else {
+        target[key] = new Array(
+          ...target[key] ?? [],
+          ...value,
+        );
+      }
```

Comment:
I would like the logic to have positive paths with the `else` as the catch all. So if key === 'command', follow the new behavior (don't dedup), otherwise keep the behavior the same.